modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/permission/DefaultSearchResultPermissionFilter.java

```diff
@@ -335,8 +343,54 @@ private boolean _isIncludeDocument(
 		return false;
 	}
 
+	private void _updateSearchHits(Hits hits, SearchContext searchContext) {
+		SearchResponseImpl searchResponseImpl =
+			(SearchResponseImpl)searchContext.getAttribute("search.response");
+
+		if (searchResponseImpl == null) {
+			return;
+		}
+
+		SearchHits searchHits = searchResponseImpl.getSearchHits();
+
+		List<SearchHit> searchHitsList = searchHits.getSearchHits();
+
+		if (searchHitsList.isEmpty()) {
+			return;
```

Comment:
just confirming its intended to have no `searchResponseImpl.setSearchHits(searchHitsBuilder.build());` here?

Reply:
Yes, on line 371 there are searchHits that need to be removed, but here the searchHits are already empty.